test/e2e/persistent_volumes.go

```diff
@@ -374,17 +400,22 @@ func testPodSuccessOrFail(c clientset.Interface, ns string, pod *api.Pod) {
 
 // Delete the passed in pod.
 func deletePod(f *framework.Framework, c clientset.Interface, ns string, pod *api.Pod) {
+	if c != nil {
+		if pod != nil && len(pod.Name) > 0 {
+			framework.Logf("Deleting pod %v", pod.Name)
+			_, err := c.Core().Pods(ns).Get(pod.Name)
+			if !apierrs.IsNotFound(err) {
+				Expect(err).NotTo(HaveOccurred())
+				err := c.Core().Pods(ns).Delete(pod.Name, nil)
+				Expect(err).NotTo(HaveOccurred())
 
-	framework.Logf("Deleting pod %v", pod.Name)
-	err := c.Core().Pods(ns).Delete(pod.Name, nil)
-	Expect(err).NotTo(HaveOccurred())
-
-	// Wait for pod to terminate.  Expect apierr NotFound
-	err = f.WaitForPodTerminated(pod.Name, "")
-	Expect(err).To(HaveOccurred())
-	Expect(apierrs.IsNotFound(err)).To(BeTrue())
-
-	framework.Logf("Ignore \"not found\" error above. Pod %v successfully deleted", pod.Name)
+				// Wait for pod to terminate.  Expect apierr NotFound
+				err = f.WaitForPodTerminated(pod.Name, "")
+				Expect(err).To(HaveOccurred())
+				framework.Logf("Ignore \"not found\" error above. Pod %v successfully deleted", pod.Name)
```

Comment:
@jeffvance I traced it down and it does one of 3 errors:
1. is not found
2. custom error on api.PodFailed == true
3. timeout